bika/lims/subscribers/dep_cookie.py

```diff
@@ -87,7 +87,7 @@ def SetDepartmentCookies(event):
             elif brain[0].portal_type == 'LabContact':
                 lab_con = brain[0].getObject()
                 if lab_con.getDefaultDepartment():
-                    dep_for_cookie = lab_con.getDefaultDepartment()
+                    dep_for_cookie = lab_con.getDefaultDepartment().UID()
```

Comment:
Better to use functions from the `api`:

```python
lab_con = api.get_object(brain[0])
department= lab_con.getDefaultDepartment()
if department:
    dep_for_cookie = api.get_uid(department)
```

BTW, I suggest to review this whole function `SetDepartmentCookies`. Too long and I feel its complexity can be reduced a lot.